src/app/handlers/auto_unacknowledge.handler.ts

```diff
@@ -0,0 +1,13 @@
+import { GuildMember } from 'discord.js';
+import { IContainer, IHandler } from '../../common/types';
+
+export class AutoAcknowledgeHandler implements IHandler {
+  constructor(public container: IContainer) {}
+
+  public async execute(member: GuildMember): Promise<void> {
+    const unacknowledgedRole = member.guild.roles.filter(
+      (role) => role.name.toLowerCase() === 'unacknowledged'
+    );
```

Comment:
can we do it in constructor to save it?

Reply:
Bot Crashes when i try that

Reply:
okay then consider calcing it here but then saving the result

Reply:
Bot crashes
`Property '_unacknowledgedRole' has no initializer and is not definitely assigned in the constructor`

Reply:
could change to
`const unacknowledgedRole = member.guild.roles.get('unacknowledged');`

Reply:
nvm, doesn't work

Reply:
> Property '_unacknowledgedRole' has no initializer and is not definitely assigned in the constructor

means you have to initialise it to `undefined` at the very least. Then in the `execute` you check if `undefined`, and if so load the role; otherwise continue